### PR TITLE
feat(dtfs2-6966): add audit logs to update facility status endpoint

### PR DESCRIPTION
**Pull request by AlexBramhill**
## Introduction :pencil2:
We need to add audit details to portal endpoints

## Resolution :heavy_check_mark:
This PR adds audit details to the `update facility status` endpoint

### CODE REVIEW

File: dtfs-central-api/src/v1/controllers/portal/facility/update-facility-status.controller.js [hunk 1]
```diff
@@ -41,11 +45,11 @@ exports.updateFacilityStatusPut = async (req, res) => {
   if (ObjectId.isValid(req.params.id)) {
     const facilityId = req.params.id;
 
-    const { status } = req.body;
+    const { status, auditDetails } = req.body;
```
Comment (MarRobSoftwire) [inline dtfs-central-api/src/v1/controllers/portal/facility/update-facility-status.controller.js:48]: Need to validate auditDetails

Comment (MarRobSoftwire) [reply dtfs-central-api/src/v1/controllers/portal/facility/update-facility-status.controller.js:48]: and fwiw can do your usual destructuring of `req`

File: dtfs-central-api/src/v1/controllers/portal/facility/update-facility-status.controller.js [hunk 2]
```diff
@@ -10,18 +11,21 @@ const withoutId = (obj) => {
   return cleanedObject;
 };
 
-const updateFacilityStatus = async (facilityId, status, existingFacility) => {
+const updateFacilityStatus = async (facilityId, status, existingFacility, auditDetails) => {
```
Comment (MarRobSoftwire) [inline dtfs-central-api/src/v1/controllers/portal/facility/update-facility-status.controller.js:14]:
```suggestion
const updateFacilityStatus = async ({facilityId, status, existingFacility, auditDetails}) => {
```

File: dtfs-central-api/src/v1/routes/portal-routes.js [hunk 3]
```diff
@@ -455,9 +455,11 @@ portalRouter.route('/facilities/:id').delete(deleteFacilityController.deleteFaci
...
  *       content:
  *         application/json:
- *           schema:
- *             type: object
- *             example: { status: Ready for Checker's approval }
+ *           properties:
+ *             status:
+ *               type: string
+ *             auditDetails:
+ *               $ref: '#/definitions/portalAuditDetails'
```
Comment (MarRobSoftwire) [inline dtfs-central-api/src/v1/routes/portal-routes.js:462]:
```suggestion
 *             auditDetails:
 *               type: object
 *               $ref: '#/definitions/portalAuditDetails'
```

**Review by ChristianMcc — APPROVED** [review]
LGTM :)

**Pull request merged**